mlir/lib/Dialect/XeGPU/IR/XeGPUOps.cpp

```diff
@@ -149,8 +162,9 @@ isValidGatherScatterBufferParams(Type maskTy, VectorType valueTy,
       return emitError()
              << "Mask should match value except the chunk size dim.";
   }
-
   llvm::SmallVector<int64_t> expectedMaskShape(valueShape);
+  if (maskSize == 1)
+    return success();
```

Comment:
nit: it might be less error-prone to perform remaining checks when `maskSize` is non-unit instead of an early exist.
As in:
```
if (maskSize != 1) {
  // perform checks
}
```